### PR TITLE
Use image builders

### PR DESCRIPTION
**Pull request by dwertent**
## Overview
* Use a builder for go build
* Update some packages dependencies

## Additional Information
The nodeAgent image vulnerability scan **before** this PR:
```
% quay.io/kubescape/node-agent:v0.0.63
0 critical, 1 high, 69 medium, 83 low, 28 negligible
```

```
% node-agent
0 critical, 3 high, 1 medium, 0 low, 0 negligible
```

### CODE REVIEW

**Comment by dwertent** [comment]
Two vulnerabilities have available fixed versions:
```
% grype node-agent

NAME        INSTALLED  FIXED-IN  TYPE  VULNERABILITY  SEVERITY
libcrypto3  3.1.0-r4   3.1.1-r0  apk   CVE-2023-2650  High
libssl3     3.1.0-r4   3.1.1-r0  apk   CVE-2023-2650  High
```

@rcohencyberarmor we should find some time to fix them.

File: build/Dockerfile [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
+FROM quay.io/kubescape/ebpf-engine:v1.0.0 as base
 
+FROM golang:1.20-alpine as builder
 
-FROM ubuntu:latest
+ENV GO111MODULE=on 
+ENV CGO_ENABLED=0
 
-RUN apt update && apt-get install git libelf-dev golang-go -y
+RUN mkdir /etc/node-agent
```
Comment (matthyx) [inline build/Dockerfile:8]: why building in /etc?

Comment (dwertent) [reply build/Dockerfile:8]: I dont know. This is what was in the previous Dockerfile and I do not know what dependencies we need from there.
We can have this sorted out in a different PR.

Comment (matthyx) [reply build/Dockerfile:8]: ok, when we make the move to efficient multi-platform builds 👍

**Review by matthyx — APPROVED** [review]

**Comment by dwertent** [comment]
@matthyx I'll need you to approve the PR again

**Review by matthyx — APPROVED** [review]

**Pull request merged**